Makefile

```diff
@@ -30,7 +30,7 @@ PYFORMAT=$(TOOLS)/pyformat
 RSTLINT=$(TOOLS)/rst-lint
 
 BROKEN_VIRTUALENV=$(BUILD_RUNTIMES)/virtualenvs/broken
-TOOL_VIRTUALENV=$(BUILD_RUNTIMES)/virtualenvs/tools
+TOOL_VIRTUALENV:=$(BUILD_RUNTIMES)/virtualenvs/tools-$(shell sha1sum requirements/tools.txt | head -c 10)
```

Comment:
(I don't care about Windows here - the build is not supported natively on Windows - but I do care about OSX)